### PR TITLE
Swiping viewport itwinviewer

### PR DESCRIPTION
**Pull request by dustinlebsock**

### CODE REVIEW

File: src/frontend-samples/swiping-viewport/SwipingComparisonApi.ts [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+import { ClipPlane, ClipPrimitive, ClipVector, ConvexClipPlaneSet, Point3d, Transform, Vector3d } from "@bentley/geometry-core";
+import { ContextRealityModelProps, FeatureAppearance, Frustum, RenderMode, ViewFlagOverrides } from "@bentley/imodeljs-common";
+import { AccuDrawHintBuilder, FeatureSymbology, GraphicBranch, IModelApp, IModelConnection, queryRealityData, RenderClipVolume, SceneContext, ScreenViewport, TiledGraphicsProvider, TileTreeReference, Viewport } from "@bentley/imodeljs-frontend";
+
+export enum ComparisonType {
+  Wireframe,
+  RealityData,
+}
+
+export default class SwipingViewportApp {
```
Comment (Ellord207) [inline src/frontend-samples/swiping-viewport/SwipingComparisonApi.ts:16]: File names is SwipingComparisonApi and class name is SwipingViewportApp  Please make these agree.

**Review by Ellord207 — CHANGES_REQUESTED** [review]

File: src/frontend-samples/swiping-viewport/SwipingComparisonApi.ts (hunk 1 above)
Comment (dustinlebsock) [reply src/frontend-samples/swiping-viewport/SwipingComparisonApi.ts]: fixed

File: src/frontend-samples/swiping-viewport/SwipingComparisonApi.ts [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+import { ClipPlane, ClipPrimitive, ClipVector, ConvexClipPlaneSet, Point3d, Transform, Vector3d } from "@bentley/geometry-core";
+import { ContextRealityModelProps, FeatureAppearance, Frustum, RenderMode, ViewFlagOverrides } from "@bentley/imodeljs-common";
+import { AccuDrawHintBuilder, FeatureSymbology, GraphicBranch, IModelApp, IModelConnection, queryRealityData, RenderClipVolume, SceneContext, ScreenViewport, TiledGraphicsProvider, TileTreeReference, Viewport } from "@bentley/imodeljs-frontend";
+
+export enum ComparisonType {
+  Wireframe,
+  RealityData,
+}
+
+export default class SwipingViewportApi {
```
Comment (williamkbentley) [inline src/frontend-samples/swiping-viewport/SwipingComparisonApi.ts:16]: Rename to `SwipingCompairsonApi`

File: src/frontend-samples/swiping-viewport/SwipingComparisonApp.tsx [hunk 3]
```diff
@@ -2,260 +2,85 @@
...
+import { IModelApp, IModelConnection, ScreenViewport } from "@bentley/imodeljs-frontend";
+import { IModelViewportControlOptions } from "@bentley/ui-framework";
+import { SwipingComparisonWidgetProvider } from "./SwipingComparisonWidget";
+import { DividerComponent } from "./Divider";
+import SwipingComparisonApi from "./SwipingComparisonApi";
+
+const uiProviders = [new SwipingComparisonWidgetProvider()];
+
+const SwipingComparisonApp: FunctionComponent = () => {
+  const sampleIModelInfo = useSampleWidget("Use the controls below to change the view attributes.", [SampleIModels.ExtonCampus]);
```
Comment (williamkbentley) [inline src/frontend-samples/swiping-viewport/SwipingComparisonApp.tsx:17]:
```suggestion
  const sampleIModelInfo = useSampleWidget("Drag the divider to compare the two halves of the view. Try rotating the view with the 'Lock Plane' toggle on and off.", [SampleIModels.ExtonCampus]);
```

File: src/frontend-samples/swiping-viewport/SwipingComparisonWidget.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    SwipingComparisonApi.attachRealityData(vp, iModelConnection!).then().catch();
+    SwipingComparisonApi.setRealityModelTransparent(vp, ComparisonType.RealityData !== comparisonState);
+    const dividerPos = initPositionDivider(SwipingComparisonApi.getClientRect(vp));
+    setDividerLeftState(dividerPos);
+    setIsInit(true);
+  }, [comparisonState, iModelConnection]);
+
+  useEffect(() => {
+    if (viewport && !isInit) {
+      /** Initally the widget is docked to the right of the screen, which is then updated to be floating in the sample showcase.
```
Comment (williamkbentley) [inline src/frontend-samples/swiping-viewport/SwipingComparisonWidget.tsx:40]:
```suggestion
      /** Initially the widget is docked to the right of the screen, which is then updated to be floating in the sample showcase.
```

File: src/frontend-samples/swiping-viewport/SwipingComparisonApp.tsx [hunk 5]
```diff
@@ -2,260 +2,85 @@
...
+  const initPositionDivider = (bounds: ClientRect): number => {
+    return bounds.left + (bounds.width / 2);
+  };
+
+  const _onDividerMoved = (leftWidth: number, rightWidth: number) => {
+    // leftWidth is relative to the canvas.  We need to track left based on the window
+    const sliderWidth = boundsState!.width - (leftWidth + rightWidth);
+    const left = leftWidth + (sliderWidth / 2);
+
+    const updatedLeft = left + boundsState!.left;
```
Comment (williamkbentley) [inline src/frontend-samples/swiping-viewport/SwipingComparisonApp.tsx:50]: nit - remove blank newline

**Review by williamkbentley — CHANGES_REQUESTED** [review]
- Update readme `TiledGraphisProvider` is no longer beta

File: src/frontend-samples/swiping-viewport/SwipingComparisonApi.ts (hunk 2 above)
Comment (dustinlebsock) [reply src/frontend-samples/swiping-viewport/SwipingComparisonApi.ts]: Renamed.

File: src/frontend-samples/swiping-viewport/SwipingComparisonApp.tsx (hunk 3 above)
Comment (dustinlebsock) [reply src/frontend-samples/swiping-viewport/SwipingComparisonApp.tsx]: Fixed.

File: src/frontend-samples/swiping-viewport/SwipingComparisonWidget.tsx (hunk 4 above)
Comment (dustinlebsock) [reply src/frontend-samples/swiping-viewport/SwipingComparisonWidget.tsx]: I suck at spelling.

File: src/frontend-samples/swiping-viewport/SwipingComparisonApp.tsx (hunk 5 above)
Comment (dustinlebsock) [reply src/frontend-samples/swiping-viewport/SwipingComparisonApp.tsx:49]: removed.

**Comment by dustinlebsock** [comment]
Removed the beta info from Readme.

File: src/frontend-samples/swiping-viewport/SwipingComparisonWidget.tsx (hunk 4 above)
Comment (williamkbentley) [reply src/frontend-samples/swiping-viewport/SwipingComparisonWidget.tsx]: Especially this particular word 😀

**Review by williamkbentley — APPROVED** [review]
LGTM

**Review by Ellord207 — APPROVED** [review]

**Pull request merged**